Development

imamfzn opened the pull request.

imamfzn committed:
Add service test

imamfzn committed:
Add recipe test

imamfzn committed:
Add recipe

imamfzn committed:
Add repository interface

imamfzn committed:
Add service

imamfzn committed:
Add xorm column definition

imamfzn committed:
Add documentation

imamfzn committed:
Add documentation

imamfzn committed:
Add documentation

imamfzn committed:
Ignore output from test coverage

imamfzn committed:
Add repository test

imamfzn committed:
Add repository

imamfzn committed:
Add env sample

imamfzn committed:
Separate package for test

imamfzn committed:
Fix method name

imamfzn committed:
Update readme

imamfzn committed:
Fix spacing

imamfzn committed:
ignore vendor

imamfzn committed:
Add vendor

imamfzn committed:
Update readme

imamfzn committed:
Add load env mechanism

SetyaK commented:
Gunakan `golint` untuk memeriksa linter.
Dependency management gunakan `go dep`, jika pakai go 1.11 gunakan `go mod`.

SetyaK commented on database/repository.go:
```diff
@@ -0,0 +1,81 @@
+func NewRepository() (bukaresep.Repository, error) {
+	gotenv.Load()
+
+	db, err := xorm.NewEngine(DEFAULT_DB_DRIVER, os.Getenv("BUKARESEP_DB_FILENAME"))
```
untuk package, variable nama database ini masukkan ke dalam structnya.
pengambilan variable dari env sebaiknya ada di dalam app saja, bukan di package

SetyaK commented on repository.go:
```diff
@@ -0,0 +1,19 @@
+package bukaresep
+
+// Repository is an interface as data access to a persistent storage
+type Repository interface {
```
file interface ini sebaiknya dipindah ke dalam direktori `repository`

imamfzn replied:
mas untuk yang ini, kenapa perlu dipindah ya? terus kalau dipindah apakah nama package nya perlu berubah? misal jadi repository

SetyaK commented on database/repository.go:
```diff
@@ -0,0 +1,81 @@
+
+const DEFAULT_DB_DRIVER = "sqlite3"
+
+type repository struct {
```
File ini sebaiknya berada dalam package yang sama dengan interfacenya.
Penamaan file dan struct disesuaikan dengan impelementasinya, misal `sqlite` (implementasi repository yang menggunakan sqlite).

imamfzn committed:
Accepting golint

Accepting golint

imamfzn committed:
Repository impelementation refactor

- Accepting golint
- Make sqlite implementation as filename and struct name

imamfzn committed:
Remove unused dependency

Remove unused dependency

imamfzn committed:
Go mod as dependency management

Using go modules as dependency management

imamfzn committed:
Update readme

Update readme

imamfzn committed:
Update readme

Update readme

imamfzn committed:
Fully code refactor

Separate repository, database and entity as package

imamfzn committed:
Update readme

Update readme

imamfzn committed:
Update readme

Update readme

SetyaK commented on repository/xorm_repository.go:
```diff
@@ -0,0 +1,61 @@
+
+// NewRepository will return an implementation of Repository.
+// It will use xorm driver implementation.
+func NewRepository(db *xorm.Engine) (Repository, error) {
```
bukankah seharusnya NewXormRepository()? karena return xormRepository

SetyaK commented on database/sqlite_database.go:
```diff
@@ -0,0 +1,25 @@
+	"github.com/imamfzn/bukaresep-go/entity"
+
+	// it required for xorm
+	_ "github.com/mattn/go-sqlite3"
```
side effect import sebaiknya digunakan di package main saja

SetyaK commented on database/sqlite_database.go:
```diff
@@ -0,0 +1,25 @@
+// CreateDatabase will create an implementation of xorm database
+// using sqlite3 driver
+func CreateDatabase(dbFilename string) (*xorm.Engine, error) {
+	db, err := xorm.NewEngine(dbDriver, dbFilename)
```
deklarasi newEngine dipindah ke main pada project webservice saja karena membutuhkan side effect import.
jadi package database ini tidak diperlukan

imamfzn committed:
Change method name

Change method name

imamfzn committed:
ChChange method name & change db instantiate

Change method name & change db instantiate

imamfzn committed:
Delete unecessary package

instantiation of database will execute in main of app / service

imamfzn committed:
Add json transform

Add json transform

imamfzn committed:
Add ToJSON test

Add ToJSON test

imamfzn committed:
Update readme

Update readme

SetyaK commented on README.md:
````diff
@@ -1,2 +1,177 @@
+```bash
+git clone -b development https://github.com/imamfzn/bukaresep-go.git
+```
+2. Install required go packages
````
tidak perlu menginstall package yg ada di development ini karena sudah menggunakan go module, dependency akan diinstall otomatis pada saat test/build

SetyaK commented on README.md:
```diff
@@ -1,2 +1,177 @@
+### Prerequisite
+
+1. Git
+2. Golang
```
perlu disebutkan juga golang versi brp, misal `>1.11`

SetyaK commented on vendor/vendor.json:
```diff
@@ -0,0 +1,19 @@
+{
```
ini file sisa govendor, sudah tida diperlukan lagi

SetyaK commented on env.sample:
```diff
@@ -0,0 +1 @@
+BUKARESEP_DB_FILENAME=bukaresep
```
unused file

SetyaK commented on repository/xorm_repository.go:
```diff
@@ -0,0 +1,61 @@
+
+// NewXormRepository will return an implementation of Repository.
+// It will use xorm driver implementation.
+func NewXormRepository(db *xorm.Engine) (Repository, error) {
```
method ini perlu dibuat juga unit testnya

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+		t.Fail()
+	}
+
+	_, err = repo.Add(&entity.Recipe{Name: "Food-name", Description: "food-desc", Ingredients: "food-ings", Instructions: "food-instr"})
```
entity value hasil addnya juga perlu dicek

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+		t.Fail()
+	}
+
+	sampleRecipe, err := repo.Add(&entity.Recipe{Name: "Food-name", Description: "food-desc", Ingredients: "food-ings", Instructions: "food-instr"})
```
pastikan entity yg dikembalikan Get() sesuai dengan yg diharapkan

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+		}
+	}
+
+	recipes, err := repo.GetAll()
```
entity valuenya juga perlu dicek

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+	numOfSample := 3
+
+	for i := 0; i < numOfSample; i++ {
+		_, err = repo.Add(&entity.Recipe{Name: "Food-name", Description: "food-desc", Ingredients: "food-ings", Instructions: "food-instr"})
```
sebaiknya recipenya berbeda2 untuk dites valuenya masing2.
entity.Recipe ini dijadikan variable test pacakage saja, supaya bisa digunakan di semua unit test tanpa deklarasi ulang di tiap method.

imamfzn replied:
kalau yang ini maksudnya dibikin kayak fixture gitu bukan ya?

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+		t.Fail()
+	}
+
+	if sampleRecipe.Name != updatedName {
```
entity value yg lain juga harus dicek apakah valuenya masih tetap sama

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+
+	err = repo.Delete(sampleRecipe)
+
+	if err != nil {
```
perlu dicek juga menggunakan get all, untuk memastikan recipenya terhapus

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,166 @@
+		t.Fail()
+	}
+
+	sampleRecipe, err := repo.Add(&entity.Recipe{Name: "Food-name", Description: "food-desc", Ingredients: "food-ings", Instructions: "food-instr"})
```
tambahkan recipe lain untuk memastikan delete ini hanya menghapus recipe yg diset di parameter saja

imamfzn committed:
Remove unused file

Remove unused file

imamfzn committed:
Fix repository unit test

- Create test helper for repository
- Create entity sample as variable
- Checking each value on entity after each operation in repository

imamfzn committed:
Update readme

Update readme: add golang version

imamfzn committed:
Remove unused file

Remove unused file

imamfzn committed:
Remove unnecessary step

Remove unnecessary step

SetyaK commented:
gunakan `go vet ./...` untuk pengecekan constructsnya

SetyaK commented on repository/xorm_repository_test.go:
```diff
@@ -0,0 +1,260 @@
+	db, err := CreateDatabase()
+
+	if err != nil {
+		t.Errorf("Errow while creating a database from xorm")
```
banyak typo "Errow"

imamfzn committed:
Fix typo and code structure

- Accepting go vet
- Fix typo (comment, stderr / stdout)

imamfzn committed:
Fix typo

Fix typo
